### PR TITLE
Revert "Revert "Un-flake test_undrop_query""

### LINKED ISSUE #55167: Flaky test_undrop_query
https://s3.amazonaws.com/clickhouse-test-reports/0/0629b8679b934e38477e67bd07d319498cd496d1/integration_tests__tsan__[2_6].html

```
E           helpers.client.QueryRuntimeException: Client failed! Return code: 60, stderr: Received exception from server (version 23.9.1):
E           Code: 60. DB::Exception: Received from 172.16.15.2:9000. DB::Exception: The drop task of table default.test_undrop_loop6 (a7aa9db2-5bb7-11ee-8786-0242ac110002) is in progress, has been dropped or the database engine doesn't support it. Stack trace:
E           
E           0. ./build_docker/./contrib/llvm-project/libcxx/include/exception:134: Poco::Exception::Exception(String const&, int) @ 0x000000001f986ba3 in /usr/bin/clickhouse
E           1. ./build_docker/./src/Common/Exception.cpp:98: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000fb86154 in /usr/bin/clickhouse
E           2. DB::Exception::Exception<String>(int, FormatStringHelperImpl<std::type_identity<String>::type>, String&&) @ 0x0000000007dc16f9 in /usr/bin/clickhouse
E           3. ./build_docker/./src/Interpreters/DatabaseCatalog.cpp:0: DB::DatabaseCatalog::dequeueDroppedTableCleanup(DB::StorageID) @ 0x0000000018c64d1c in /usr/bin/clickhouse
E           4. ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: DB::InterpreterUndropQuery::executeToTable(DB::ASTUndropQuery&) @ 0x000000001996ef47 in /usr/bin/clickhouse
E           5. ./build_docker/./src/Interpreters/InterpreterUndropQuery.cpp:44: DB::InterpreterUndropQuery::execute() @ 0x000000001996e6bb in /usr/bin/clickhouse
E           6. ./build_docker/./src/Interpreters/executeQuery.cpp:0: DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x0000000019da760b in /usr/bin/clickhouse
E           7. ./build_docker/./src/Interpreters/executeQuery.cpp:1234: DB::executeQuery(String const&, std::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x0000000019da30b9 in /usr/bin/clickhouse
E           8. ./build_docker/./src/Server/TCPHandler.cpp:0: DB::TCPHandler::runImpl() @ 0x000000001b267d53 in /usr/bin/clickhouse
E           9. ./build_docker/./src/Server/TCPHandler.cpp:2127: DB::TCPHandler::run() @ 0x000000001b281d48 in /usr/bin/clickhouse
E           10. ./build_docker/./base/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0x000000001f802a03 in /usr/bin/clickhouse
E           11. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:48: Poco::Net::TCPServerDispatcher::run() @ 0x000000001f80327b in /usr/bin/clickhouse
E           12. ./build_docker/./base/poco/Foundation/src/ThreadPool.cpp:202: Poco::PooledThread::run() @ 0x000000001fa141a7 in /usr/bin/clickhouse
E           13. ./build_docker/./base/poco/Foundation/src/Thread.cpp:46: Poco::(anonymous namespace)::RunnableHolder::run() @ 0x000000001fa12470 in /usr/bin/clickhouse
E           14. ./build_docker/./base/poco/Foundation/include/Poco/SharedPtr.h:231: Poco::ThreadImpl::runnableEntry(void*) @ 0x000000001fa1092a in /usr/bin/clickhouse
E           15. __tsan_thread_start_func @ 0x0000000007d2dc8f in /usr/bin/clickhouse
E           16. ? @ 0x00007f3e4fdb5b43 in ?
E           17. ? @ 0x00007f3e4fe47a00 in ?
E           . (UNKNOWN_TABLE)
E           (query: undrop table test_undrop_loop6 uuid 'a7aa9db2-5bb7-11ee-8786-0242ac110002' settings allow_experimental_undrop_table_query = 1;)
```

### PR DESCRIPTION
**Pull request by yariks5s**
Fix https://github.com/ClickHouse/ClickHouse/issues/55167 (improvement of #61348 by @rschu1ze)

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 6a19b77d9ea4545bf7080896a760022ed8fd9410 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61727/6a19b77d9ea4545bf7080896a760022ed8fd9410/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 6a19b77d9ea4545bf7080896a760022ed8fd9410 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61727/6a19b77d9ea4545bf7080896a760022ed8fd9410/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/61727/6a19b77d9ea4545bf7080896a760022ed8fd9410/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/02681_undrop_query.sql [hunk 1]
```diff
@@ -85,5 +85,5 @@ drop table 02681_undrop_multiple;
 select table from system.dropped_tables where table = '02681_undrop_multiple' limit 1;
 undrop table 02681_undrop_multiple;
 select * from 02681_undrop_multiple order by id;
-undrop table 02681_undrop_multiple; -- { serverError 57 }
+undrop table 02681_undrop_multiple; -- { serverError TABLE_ALREADY_EXISTS }
```
Comment (rschu1ze) [inline tests/queries/0_stateless/02681_undrop_query.sql:88]: While at it, you could do the same in l. 20.

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 2]
```diff
@@ -945,9 +945,9 @@ Hard limit is configured via system tools
 
 ## database_atomic_delay_before_drop_table_sec {#database_atomic_delay_before_drop_table_sec}
 
-Sets the delay before remove table data in seconds. If the query has `SYNC` modifier, this setting is ignored.
+The delay before a table data is dropped in seconds. If the `DROP TABLE` query has a `SYNC` modifier, this setting is ignored.
```
Comment (rschu1ze) [inline docs/en/operations/server-configuration-parameters/settings.md:948]: Someone who doesn't know the UNDROP statement may be confused if DROP always incurs a (synchronous) delay of 8 minutes. Maybe rephrase like this?

```
The delay during which a dropped table can be restored using the UNDROP statement. If `DROP TABLE` ran with a `SYNC` modifier, the setting is ignored.
```

(ideally, you link UNDROP with https://clickhouse.com/docs/en/sql-reference/statements/undrop)

File: tests/integration/test_undrop_query/configs/with_delay_config.xml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <clickhouse>
-    <database_atomic_delay_before_drop_table_sec>3</database_atomic_delay_before_drop_table_sec>
+    <database_atomic_delay_before_drop_table_sec>80</database_atomic_delay_before_drop_table_sec>
```
Comment (rschu1ze) [inline tests/integration/test_undrop_query/configs/with_delay_config.xml:2]: Okay, a too-small drop time seems the culprit, thanks for finding it.

File: tests/integration/test_undrop_query/test.py [hunk 4]
```diff
@@ -20,29 +21,25 @@ def started_cluster():
 
 
 def test_undrop_drop_and_undrop_loop(started_cluster):
-    # create, drop, undrop, drop, undrop table 5 times
-    for _ in range(5):
-        table_uuid = str(uuid.uuid1())
-        table = f"test_undrop_loop"
+    uuid_list = []
```
Comment (rschu1ze) [inline tests/integration/test_undrop_query/test.py:24]: The test IMHO runs too long for the stuff it tests. What about this:

- set the drop time setting to 20 sec (XML file).
- create four tables
- drop a table, undrop it, sleep 5 sec (*)
- execute (*) three times in a row
- sleep 10 sec, this should place us ca. 25 sec after test start.
- UNDROP the 4th table, expect that this fails

**Comment by CheSema** [comment]
In general it is good for me, all are as we discussed.
Make the time (80s) less is a good idea. Also take in mind that all operations took some time to run, so you schedule could shift in some way. If you are going to reduce run time, you may need to sleep less than constant: make some schedule in advance and sleep until some point in time.

**Review by CheSema — APPROVED** [review]

**Review by rschu1ze — APPROVED** [review]

**Review by rschu1ze — APPROVED** [review]
